enforcer/src/policy/policy_import.c

```diff
@@ -506,12 +506,13 @@ int policy_import(int sockfd, engine_type* engine, db_connection_t *dbconn,
             continue;
         }
 
-        if (p->denial_salt_length != (xpolicies+i)->denial_salt_length)
-            resalt_task_flush(engine, dbconn, p->name);
-
         xml2db(p, xpolicies+i);
-        if (!p->scratch)
+        if (!p->scratch) {
             p->scratch = 3; /* do not delete */
+            if (p->denial_salt_length != (xpolicies+i)->denial_salt_length) {
+                resalt_task_flush(engine, dbconn, p->name);
```

Comment:
Good idea. However the new length isn't committed to the database at this point. So the resalt task probably does not do what you want.

Reply:
Correct